From 404e10ce0defe1950770e3e01f5a246cf8e8e4d4 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 20 Mar 2020 04:36:03 -0700
Subject: [PATCH 1285/1851] clk: idt: Fix i2cwritebulk() failure

commit a6f1a39da81b3895105a7c543a9e69de7fbfebe7 from
https://github.com/Xilinx/linux-xlnx.git

Commit 8b9f9d4dc511 ("regmap: verify if register is writeable before
writing operations") introduces new way for validating the writable
register range. This causes the idt driver to fail as max_register is
only 0xFF and reg address falls outside it. This works earlier as the
writeable_reg() function pointer to idt24x_regmap_is_writeable() is used
which always returns true.

So to pass the validation, the max_register is now set to 0xFFFF.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/clk/idt/clk-idt8t49n24x.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/idt/clk-idt8t49n24x.c b/drivers/clk/idt/clk-idt8t49n24x.c
index 79289a5a1934..878637986a60 100644
--- a/drivers/clk/idt/clk-idt8t49n24x.c
+++ b/drivers/clk/idt/clk-idt8t49n24x.c
@@ -346,7 +346,7 @@ static const struct regmap_config idt24x_regmap_config = {
 	.reg_bits = 16,
 	.val_bits = 8,
 	.cache_type = REGCACHE_RBTREE,
-	.max_register = 0xff,
+	.max_register = 0xffff,
 	.writeable_reg = idt24x_regmap_is_writeable,
 	.volatile_reg = idt24x_regmap_is_volatile,
 };
-- 
2.31.1

